tools/experimental/dynamic_shapes/torchfuzz/ops_fuzzer.py

```diff
@@ -188,239 +185,65 @@ def fuzz_op(target_spec: Spec, depth, stack_size) -> tuple[str, list[Spec]]:
             else:
-                return [lower_dtype, target_dtype]
+                chosen_op_name, chosen_operator = random.choice(leaf_ops) if leaf_ops else random.choice(non_leaf_ops)
         else:
-            # Fallback to same type if no lower types available
-            return [target_dtype, target_dtype]
-
-
-def _get_promoted_scalar_args(target_dtype: torch.dtype) -> list[Spec]:
-    """
-    Generate two argument specs that will promote to target_dtype via PyTorch's type promotion rules.
-    """
-    arg_dtypes = _get_promoted_dtypes(target_dtype)
-
-    # For ScalarSpec output, both inputs must be ScalarSpec
-    # (mixing with 0-D TensorSpec would produce 0-D TensorSpec output)
-    return [ScalarSpec(arg_dtypes[0]), ScalarSpec(arg_dtypes[1])]
-
-
-def _get_item_args_specs(target_spec: ScalarSpec) -> tuple[str, list[Spec]]:
-    """Get argument specifications for torch.ops.aten.item operation."""
-    # torch.ops.aten.item: tensor -> scalar (extract single element)
-    # Create a tensor spec that can produce a scalar via .item()
-    tensor_spec = TensorSpec(
-        size=(1,), stride=(1,), dtype=target_spec.dtype
-    )  # 1-D tensor with 1 element
-    arg_specs: list[Spec] = [tensor_spec]
-    return "torch.ops.aten.item", arg_specs
-
-
-def _get_aten_add_args_specs(target_spec: TensorSpec) -> tuple[str, list[Spec]]:
-    """Get argument specifications for torch.ops.aten.add operation using type promotion rules."""
-    # Use promotion rules to generate diverse tensor input types
-    arg_dtypes = _get_promoted_dtypes(target_spec.dtype)
-
-    arg_specs: list[Spec] = [
-        TensorSpec(target_spec.size, target_spec.stride, arg_dtypes[0]),
-        TensorSpec(target_spec.size, target_spec.stride, arg_dtypes[1]),
-    ]
-    return "torch.ops.aten.add", arg_specs
-
-
-def _get_aten_mul_args_specs(target_spec: TensorSpec) -> tuple[str, list[Spec]]:
-    """Get argument specifications for torch.ops.aten.mul operation using type promotion rules."""
-    # Use promotion rules to generate diverse tensor input types
-    arg_dtypes = _get_promoted_dtypes(target_spec.dtype)
-
-    arg_specs: list[Spec] = [
-        TensorSpec(target_spec.size, target_spec.stride, arg_dtypes[0]),
-        TensorSpec(target_spec.size, target_spec.stride, arg_dtypes[1]),
-    ]
-    return "torch.ops.aten.mul", arg_specs
-
-
-def _get_constant_args_specs(target_spec: Spec) -> tuple[str, list[Spec]]:
-    """Get argument specifications for constant operation."""
-    # Constant operation takes no arguments - generates a fixed constant value/tensor
-    return "constant", []
+            # Normal probability distribution
+            all_ops = non_leaf_ops + leaf_ops
+            chosen_op_name, chosen_operator = random.choice(all_ops) if all_ops else ("arg", get_operator("arg"))
+
+    # Use the operator to decompose the target spec into input specs
+    try:
+        if chosen_op_name.startswith("arg_"):
```

Comment:
if decompose (fuzz_inputs_specs) return [] for constant, arg nodes. 
this can just be 
input_specs = chosen_operator.decompose(target_spec)
return chosen_op_name, input_specs 
            
 with no branching here